packages/java-parser/src/productions/expressions.js

```diff
@@ -203,10 +203,14 @@ function defineRules($, t) {
   });
 
   $.RULE("primaryPrefix", () => {
-    let isCastExpression = false;
-    if (tokenMatcher($.LA(1).tokenType, t.LBrace)) {
-      isCastExpression = this.BACKTRACK_LOOKAHEAD($.isCastExpression);
-    }
+    const isCastExpression = $.ACTION(() => {
+      let isCastExpression = false;
+      if (tokenMatcher($.LA(1).tokenType, t.LBrace)) {
+        isCastExpression = this.BACKTRACK_LOOKAHEAD($.isCastExpression);
```

Comment:
it may suffix to only wrap
this.BACKTRACK_LOOKAHEAD here with $ACTION.
Maybe you could even put the $.ACTION inside the thisBACKTRACK_LOOKAHEAD method?